src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+
+        while(true)
+        {
+            skip=false;
+            String line=in.nextLine();
+            //if bye
+            if(line.equalsIgnoreCase("bye"))
+            {
+                break;
+            }
+            //getting index and line
+            list[i]=line;
+            String[] find_index = line.split(" ");
+
+            //Task description
+            String desc;
+
+            desc=line;
+            Task t = new Task(desc);
+
+
+            if(line.toLowerCase().contains("unmark"))
+            {
+                indexing=Integer.parseInt(find_index[1]);
+                tasks[indexing-1].markAsUnDone();
+
+
+
+                skip=true;
+                System.out.println("____________________________________________________________");
+                System.out.println("OK, I've marked this task as not done yet:\n"+
+                        "["+tasks[indexing-1].getStatusIcon()+"] "+ list[indexing-1]);
+                System.out.println("____________________________________________________________\n");
+            }
+
+
+            else if(line.toLowerCase().contains("mark"))
+            {
+                indexing=Integer.parseInt(find_index[1]);
+                tasks[indexing-1].markAsDone();
+
+
+
+                skip=true;
+                System.out.println("____________________________________________________________");
+                System.out.println("Nice! I've marked this task as done:\n"+
+                       "["+tasks[indexing-1].getStatusIcon()+"] "+ list[indexing-1]);
+                System.out.println("____________________________________________________________\n");
+
+            }
+
+            else if(line.toLowerCase().contains("todo") || line.toLowerCase().contains("deadline") || line.toLowerCase().contains("event") )
+            {
+
+
+                if(line.toLowerCase().contains("todo"))
+                {
+                    String TodoTask =line.toLowerCase().substring(5,line.length());
+                    list[i]=TodoTask;
+                    tasks[i] = new Todo(TodoTask);
+
+                }
+                else if(line.toLowerCase().contains("deadline")) {
+                    String[] ToSplitDeadline=line.split("/");
+                    String DeadlineTask =ToSplitDeadline[0].toLowerCase().substring(9,ToSplitDeadline[0].length()-1);
+                    list[i]=DeadlineTask;
+                    tasks[i] = new Deadline(DeadlineTask, ToSplitDeadline[1].substring(3,ToSplitDeadline[1].length()));
+                }
+                else if(line.toLowerCase().contains("event"))
+                {
+                    String[] ToSplitEvent=line.split("/");
+                    String EventTask =ToSplitEvent[0].toLowerCase().substring(6,ToSplitEvent[0].length());
+                    list[i]=EventTask;
+                    tasks[i] = new Event(EventTask, ToSplitEvent[1].substring(5,ToSplitEvent[1].length()),ToSplitEvent[2].substring(3,ToSplitEvent[2].length()));
```

Comment:
I have realised that your "event" and "deadline" codes do not check if the input has the keywords "from" + "to", and "by" respectively. I believe that this may lead to certain cases where it incorrectly processes inputs from the user. 
As such, I recommend using string methods such as String.matches() or String.indexOf() so as to check for the relevant keywords, and I believe that this would greatly strengthen your code's ability to handle edge cases/ different inputs.